File: app/src/lib/analysis/holding-period.ts
import type { Trade } from "@/types/database";
import { TRADE_TYPE } from "@/lib/constants/trading";

// 각 SELL trade.id → 저장된 보유 기간(일)
export function computeHoldingDays(trades: Trade[]): Map<string, number> {
  const result = new Map<string, number>();

  for (const trade of trades) {
    if (trade.trade_type === TRADE_TYPE.SELL && trade.holding_days != null) {
      result.set(trade.id, trade.holding_days);
    }
  }

  return result;
}

File: be/src/invest_note_api/domain/analysis/holding_period.py
"""저장된 보유일을 모든 SELL에 대해 dict로 반환."""
from __future__ import annotations

from typing import TYPE_CHECKING

from invest_note_api.domain.trade_types import TRADE_TYPE_SELL

if TYPE_CHECKING:
    from invest_note_api.domain.trade_types import Trade


def compute_holding_days_map(trades: list[Trade]) -> dict[str, int]:
    """각 SELL trade.id → 저장된 보유 기간(일)."""
    result: dict[str, int] = {}
    for trade in trades:
        if trade.trade_type == TRADE_TYPE_SELL and trade.holding_days is not None:
            result[trade.id] = trade.holding_days

    return result
